utils.js

```diff
@@ -6818,6 +6841,28 @@ export const extractJarArchive = function (
           version = pomProperties["version"],
           confidence = 1,
           technique = "manifest-analysis";
+        if ((!group || !name || !version) && SEARCH_MAVEN_ORG) {
+          try {
+            const sha = await checksumFile("sha1", jf);
+            const searchurl =
+              "https://search.maven.org/solrsearch/select?q=1:%22" +
+              sha +
+              "%22&rows=20&wt=json";
+            const res = await cdxgenAgent.get(searchurl, {
+              responseType: "json"
+            });
+            const data = res && res.body ? res.body["response"] : undefined;
+            if (data && data["numFound"] == 1) {
+              const jarInfo = data["docs"][0];
+              group = jarInfo["g"];
+              name = jarInfo["a"];
+              version = jarInfo["v"];
+              technique = "hash-comparison";
+            }
+          } catch (err) {
+            console.log(err);
```

Comment:
Should we add a circuit breaker where we stop after x number of errors to prevent a ddos?